src/component/tooltip/TooltipContent.js

```diff
@@ -218,12 +218,15 @@ function TooltipContent(container, api, opt) {
         }
     };
     el.onmouseleave = function () {
+        // fix #12778
+        // we should set `_inContent` before `hideLater`
+        self._inContent = false;
+
         if (self._enterable) {
             if (self._show) {
                 self.hideLater(self._hideDelay);
```

Comment:
I felt this hideLater function has a flaw. A function shall not change its behaviour when un-related variable changes.

So basically, hideLater is a trap for next developer. and I do not want to be the next one.

Reply:
It seems that we can remove the two judgements about `self._enterable` and `self._show`, `hideLater` will do it.

https://github.com/apache/incubator-echarts/blob/9a2e192c7387663d83e753dc9b8a6c7683e22630/src/component/tooltip/TooltipContent.js#L312-L324

What do you think of it?

Reply:
My humble opinion is hideLater function did not do what it claims. A function shall have a single responsibility. HideLater at the moment has two responsibilities, hence it made the code complex to read and prone to bug.